packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -18,24 +25,61 @@ class AndroidCameraCameraX extends CameraPlatform {
     CameraPlatform.instance = AndroidCameraCameraX();
   }
 
-  /// ProcessCameraProvider used to get list of cameras. Visible only for testing.
+  // Instances used to access camera functionality:
+
+  /// The [ProcessCameraProvider] instance used to access camera functionality.
   @visibleForTesting
   ProcessCameraProvider? processCameraProvider;
 
-  /// Camera selector used to determine which CameraInfos are back cameras.
+  /// The [Camera] instance returned by the [processCameraProvider] when a [UseCase] is
+  /// bound to the lifecycle of the camera it manages.
+  @visibleForTesting
+  Camera? camera;
+
+  // Instances used to configure and bind use cases to ProcessCameraProvider instance:
+
+  /// The [Preview] instance that can be instantiated adn configured to present
+  /// a live camera preview.
   @visibleForTesting
-  CameraSelector? backCameraSelector = CameraSelector.getDefaultBackCamera();
```

Comment:
I ended up refactoring this, primarily for testing reasons. Initializing it like this would cause us to have to initialize this in every test, so I apologize for noting this as an option.

To do this, I modified `availableCameras()` to use a helper method to create `CameraSelector`s that is stubbed in the mock version of `AndroidCameraCameraX` that I created for testing. Let me know what you think!

Reply:
I think that makes sense! 

It means we will have to recreate those camera selectors for each call to availableCameras, but I don't think there is much reason to repeatedly make calls to availableCameras. 

It also looks like it makes the testing much cleaner.